### PR TITLE
coq_makefile: use DELETE_ON_ERROR

### PR DESCRIPTION
**Pull request by SkySkimmer**
https://www.gnu.org/software/make/manual/make.html#index-_002eDELETE_005fON_005fERROR-1

### CODE REVIEW

**Comment by SkySkimmer** [comment]
@JasonGross said on zulip:
>There's a similar issue in coq_makefile, where having native compilation fail with a stack overflow does not result in the .vo file being deleted, and so the build results get out of sync.

This should help.

**Review by gares — APPROVED** [review]

**Comment by gares** [comment]
IIRC you can also use .PRECIOUS to keep only some files (if you know their names). I don't know how much temporary cruft is there actually

**Review by JasonGross — APPROVED** [review]

**Review by JasonGross — CHANGES_REQUESTED** [review]
This deserves a changeling entry, no?  Otherwise, LGTM

**Comment by ejgallego** [comment]
Note that for some native targets this will not be enough due to https://github.com/coq/coq/issues/13035

**Comment by ejgallego** [comment]
This need rework due to #14240 , right @SkySkimmer ?

**Comment by SkySkimmer** [comment]
no

**Comment by ejgallego** [comment]
> no

How come, you added the `DELETE_ON_ERROR` there, so what are the semantics of this PR after it?

**Comment by SkySkimmer** [comment]
One is legacy build, the other is coq_makefile
no interaction

**Comment by ejgallego** [comment]
> One is legacy build, the other is coq_makefile
> no interaction

oups, thanks!

**Review by JasonGross — APPROVED** [review]

File: doc/changelog/08-cli-tools/14238-delete-error.rst [hunk 1]
```diff
@@ -0,0 +1,4 @@
+- **Changed:**
+  Makefiles produced by ``coq_makefile`` now use ``.DELETE_ON_ERROR``
```
Comment (JasonGross) [inline doc/changelog/08-cli-tools/14238-delete-error.rst:2]: Do you want to mention KEEP_ERROR here or in the reference manual?

Comment (SkySkimmer) [reply doc/changelog/08-cli-tools/14238-delete-error.rst:2]: Oh right I forgot I did that.
TBH I'm not sure it's even worth keeping, WDYT?

Comment (JasonGross) [reply doc/changelog/08-cli-tools/14238-delete-error.rst:2]: I think it's reasonable to keep, especially if people want to inspect intermediates or debug things

Comment (SkySkimmer) [reply doc/changelog/08-cli-tools/14238-delete-error.rst:2]: In the manual we say https://coq.github.io/doc/master/refman/practical-tools/utilities.html#coqmakefile-local
>The variable must belong to the variables listed in the Parameters section of the generated makefile. Here we describe only few of them.

KEEP_ERROR isn't important enough to add to the list IMO, so no need for more doc

**Comment by JasonGross** [comment]
@coqbot merge now

**Pull request merged**